src/Note.ts

```diff
@@ -49,8 +50,13 @@ class Note implements PlayableInterface {
     return 440 * power;
   }
 
-  public play(audioContext: AudioContext, oscillator: OscillatorType, resolve: () => void = () => {}): void {
-    const oscillatorNode = this.getOscillator(audioContext, oscillator);
+  public play(audioContext: AudioContext, oscillator: CustomOscillatorType, resolve: () => void = () => {}): void {
+    let oscillatorNode: OscillatorNode;
+    if(oscillator === 'piano') {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I wonder if there is a way to use a constant or anything here that is not a simple String, it feels fragile

Reply:
If the value is going to be reassigned, I don't see how we could use a const. Maybe something else I don't know...